readthedocs/donate/views.py

```diff
@@ -54,3 +56,15 @@ def get_queryset(self):
 
     def get_template_names(self):
         return [self.template_name]
+
+
+def click_proxy(request, promo_id):
```

Comment:
I believe that shouldn't require a session, as the csrftoken is a separate cookie. It should be passed by the browser as it's on the same domain. I didn't find much on implementing this on GET requests, besides purists saying "Don't have side effects".